core/src/main/java/bisq/core/btc/nodes/LocalBitcoinNode.java

```diff
@@ -24,40 +53,321 @@
+     * not BIP111-aware.
+     * Source routines and data can be found in Bitcoinj under:
+     * core/src/main/java/org/bitcoinj/core/VersionMessage.java
+     * and
+     * core/src/main/java/org/bitcoinj/core/NetworkParameters.java
+     */
+    private static boolean isBloomFilteringSupportedAndEnabled(VersionMessage versionMessage) {
+        // A service bit that denotes whether the peer supports BIP37 bloom filters or
+        // not. The service bit is defined in BIP111.
+        int NODE_BLOOM = 1 << 2;
+
+        int BLOOM_FILTERS_BIP37_PROTOCOL_VERSION = 70000;
+        var whenBloomFiltersWereIntroduced = BLOOM_FILTERS_BIP37_PROTOCOL_VERSION;
+
+        int BLOOM_FILTERS_BIP111_PROTOCOL_VERSION = 70011;
+        var whenBloomFiltersWereDisabledByDefault = BLOOM_FILTERS_BIP111_PROTOCOL_VERSION;
+
+        int clientVersion = versionMessage.clientVersion;
+        long localServices = versionMessage.localServices;
+
+        if (clientVersion >= whenBloomFiltersWereIntroduced
+                && clientVersion < whenBloomFiltersWereDisabledByDefault)
+            return true;
+        return (localServices & NODE_BLOOM) == NODE_BLOOM;
+    }
+
+    /* Performs a blocking Bitcoin protocol handshake, which includes exchanging version
+     * messages and acks. Its purpose is to check if a local Bitcoin node is running,
+     * and, if it is, check its advertised configuration. The returned Optional is empty,
+     * if a local peer wasn't found, or if handshake failed for some reason. This method
+     * could be noticably simplified, by turning connection failure callback into a
+     * future and using a first-future-to-complete type of construct, but I couldn't find
+     * a ready-made implementation.
+     */
+    private Optional<VersionMessage> attemptHandshakeForVersionMessage() {
+        Peer peer;
+        try {
+            peer = createLocalPeer(port);
+        } catch (UnknownHostException ex) {
+            log.error("Local bitcoin node handshake attempt unexpectedly threw: {}", ex);
+            return Optional.empty();
+        }
+
+        /* We temporarily silence BitcoinJ NioClient's and NioClientManager's loggers,
+         * because when a local Bitcoin node is not found they pollute console output
+         * with "connection refused" error messages.
+         */
+        var originalNioClientLoggerLevel = silence(NioClient.class);
+        var originalNioClientManagerLoggerLevel = silence(NioClientManager.class);
+
+        NioClient client;
+
+        try {
+            log.info("Initiating attempt to connect to and handshake with a local "
+                    + "Bitcoin node (which may or may not be running) on port {}.",
+                    port);
+            client = createClient(peer, port, CONNECTION_TIMEOUT);
+        } catch (IOException ex) {
+            log.error("Local bitcoin node handshake attempt unexpectedly threw: {}", ex);
```

Comment:
Using it like that the placeholder will never be filled. Maybe something like:
```suggestion
            log.error("Local bitcoin node handshake attempt was unexpectedly interrupted", ex);
```